jsoncpp: 1.9.2 -> 1.9.4

CrystalGamma opened the pull request:
Merges both patches previously applied. Also seems to change the path that the library is written to.

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://hydra.nixos.org/job/nixpkgs/trunk/manual/latest/download/1/nixpkgs/manual.html#chap-reviewing-contributions
-->

###### Motivation for this change

Fixes a build failure on platforms with unsigned `char` type, which was fixed with a patch on ARM only, but is needed on ppc64le (CC @r-burns) too. Updating to latest upstream release is probably cleaner than adding another platform test to the patch.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

Commit:
jsoncpp: 1.9.2 -> 1.9.4

Merges both patches previously applied. Also seems to change the path that the library is written to.

SuperSandro2000 commented on pkgs/development/libraries/jsoncpp/default.nix:
```diff
@@ -23,27 +23,13 @@ stdenv.mkDerivation rec {
   # CMAKE_SKIP_BUILD_RPATH to avoid cmake resetting rpath on install
   preBuild = if stdenv.isDarwin then ''
-    export DYLD_LIBRARY_PATH="`pwd`/src/lib_json''${DYLD_LIBRARY_PATH:+:}$DYLD_LIBRARY_PATH"
+    export DYLD_LIBRARY_PATH="`pwd`/lib''${DYLD_LIBRARY_PATH:+:}$DYLD_LIBRARY_PATH"
```
```suggestion
    export DYLD_LIBRARY_PATH="$(pwd)/lib''${DYLD_LIBRARY_PATH:+:}$DYLD_LIBRARY_PATH"
```

SuperSandro2000 commented on pkgs/development/libraries/jsoncpp/default.nix:
```diff
@@ -23,27 +23,13 @@ stdenv.mkDerivation rec {
+    export DYLD_LIBRARY_PATH="`pwd`/lib''${DYLD_LIBRARY_PATH:+:}$DYLD_LIBRARY_PATH"
   '' else ''
-    export LD_LIBRARY_PATH="`pwd`/src/lib_json''${LD_LIBRARY_PATH:+:}$LD_LIBRARY_PATH"
+    export LD_LIBRARY_PATH="`pwd`/lib''${LD_LIBRARY_PATH:+:}$LD_LIBRARY_PATH"
```
```suggestion
    export LD_LIBRARY_PATH="$(pwd)/lib''${LD_LIBRARY_PATH:+:}$LD_LIBRARY_PATH"
```

r-burns reviewed (approved):
Looks good, can confirm the ppc64le issue is now fixed. WFM on darwin and non-nixos linux. Might want to reduce closure size with `outputs = [ "out" "dev" ]` and `"-DBUILD_OBJECT_LIBS=OFF"`

CrystalGamma commented:
Thanks for the feedback. I went ahead and replaced the backtick expansion (can't actually quote it here because I can't get escaping the backticks to work right) by `$PWD` which is even cleaner than `$(pwd)` (IMO) and implemented the closure reduction options suggested.
